src/main/kotlin/no/nav/syfo/service/SenderFacade.kt

```diff
@@ -20,27 +18,59 @@ import no.nav.syfo.kafka.producers.dittsykefravaer.DittSykefravaerMeldingKafkaPr
 import no.nav.syfo.kafka.producers.dittsykefravaer.domain.DittSykefravaerVarsel
 
 class SenderFacade(
-    val dineSykmeldteHendelseKafkaProducer: DineSykmeldteHendelseKafkaProducer,
-    val dittSykefravaerMeldingKafkaProducer: DittSykefravaerMeldingKafkaProducer,
-    val brukernotifikasjonerService: BrukernotifikasjonerService,
-    val arbeidsgiverNotifikasjonService: ArbeidsgiverNotifikasjonService,
-    val fysiskBrevUtsendingService: FysiskBrevUtsendingService,
-    val database: DatabaseInterface
+    private val dineSykmeldteHendelseKafkaProducer: DineSykmeldteHendelseKafkaProducer,
+    private val dittSykefravaerMeldingKafkaProducer: DittSykefravaerMeldingKafkaProducer,
+    private val brukernotifikasjonerService: BrukernotifikasjonerService,
+    private val arbeidsgiverNotifikasjonService: ArbeidsgiverNotifikasjonService,
+    private val fysiskBrevUtsendingService: FysiskBrevUtsendingService,
+    val database: DatabaseInterface,
 ) {
     fun sendTilDineSykmeldte(
         varselHendelse: NarmesteLederHendelse,
         varsel: DineSykmeldteVarsel,
     ) {
-        dineSykmeldteHendelseKafkaProducer.sendVarsel(varsel)
-        lagreUtsendtNarmesteLederVarsel(DINE_SYKMELDTE, varselHendelse, varsel.id.toString())
+        var isSendingSucceed = true
+        try {
+            dineSykmeldteHendelseKafkaProducer.sendVarsel(varsel)
+        } catch (e: Exception) {
+            isSendingSucceed = false
+            log.warn("Error while sending varsel to DINE_SYKMELDTE: ${e.message}", e)
+            lagreIkkeUtsendtNarmesteLederVarsel(
+                kanal = DINE_SYKMELDTE,
+                varselHendelse = varselHendelse,
+                eksternReferanse = varsel.id.toString(),
+                feilmelding = e.message,
+                merkelapp = null,
+            )
+        }
+        if (isSendingSucceed) {
+            lagreUtsendtNarmesteLederVarsel(DINE_SYKMELDTE, varselHendelse, varsel.id.toString())
```

Comment:
Hvorfor er ikke `lagreUtsendtNarmesteLederVarsel(DINE_SYKMELDTE, varselHendelse, varsel.id.toString())` inne i try/catchen? Da slipper du å ha en boolean (isSendingSucceed) for å holde styr på det. Hvis det oppstår en exception når man kaller `dineSykmeldteHendelseKafkaProducer.sendVarsel(varsel)` så blir ikke `lagreIkkeUtsendtNarmesteLederVarsel` kalt.
```suggestion
          try {
            dineSykmeldteHendelseKafkaProducer.sendVarsel(varsel)
             lagreUtsendtNarmesteLederVarsel(DINE_SYKMELDTE, varselHendelse, varsel.id.toString())
        } catch (e: Exception) {
            isSendingSucceed = false
            log.warn("Error while sending varsel to DINE_SYKMELDTE: ${e.message}", e)
            lagreIkkeUtsendtNarmesteLederVarsel(
                kanal = DINE_SYKMELDTE,
                varselHendelse = varselHendelse,
                eksternReferanse = varsel.id.toString(),
                feilmelding = e.message,
                merkelapp = null,
            )
        }           
```

Reply:
Enig. Skal endre det samme for DITT_SYKEFRAVAER